src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java

```diff
@@ -97,11 +99,43 @@ private void applyExistingArtefact(Artefact artefact) {
     public List<Artefact> findAllByCourtId(String searchValue, Boolean verified) {
         LocalDateTime currDate = LocalDateTime.now();
         if (verified) {
-            return artefactRepository.findArtefactsBySearchVerified(searchValue, currDate);
+            return artefactRepository.findArtefactsByCourtIdVerified(searchValue, currDate);
         } else {
-            return artefactRepository.findArtefactsBySearchUnverified(searchValue, currDate);
+            return artefactRepository.findArtefactsByCourtIdUnverified(searchValue, currDate);
+        }
+    }
+
+    /**
+     * Get all relevant Artefacts based on search values stored in the Artefact.
+     *
+     * @param searchTerm the search term checking against, eg. CASE_ID or CASE_URN
+     * @param searchValue the search value to look for
+     * @param verified bool for the user being verified or not restricting the results
+     * @return list of Artefacts
+     */
+    public List<Artefact> findAllBySearch(CaseSearchTerm searchTerm, String searchValue, boolean verified) {
+        LocalDateTime currDate = LocalDateTime.now();
+        List<Artefact> artefacts;
+        switch (searchTerm) {
+            case CASE_ID:
+            case CASE_URN:
+                artefacts = verified ? artefactRepository.findArtefactBySearchVerified(searchTerm.dbValue,
+                                                                                       searchValue, currDate) :
+                    artefactRepository.findArtefactBySearchUnverified(searchTerm.dbValue, searchValue, currDate);
+                break;
+            case CASE_NAME:
+                artefacts = verified ? artefactRepository.findArtefactByCaseNameVerified(searchValue, currDate) :
+                    artefactRepository.findArtefactByCaseNameUnverified(searchValue, currDate);
+                break;
+            default:
+                throw new IllegalArgumentException(String.format("Invalid search term: %s", searchTerm));
         }
 
+        if (artefacts.isEmpty()) {
```

Comment:
Can you do in Java `if(artefacts)` or you need isEmpty check? In javaScript `if(artefact)` would evaluate to boolean (both ' ' and null would return false)

Reply:
unfortunately not in Java, its a bit more explicit in its syntax so you have to declare those kinda checks rather than relying on an undefined or empty object check also returning a bool if it contains data or not